python/examples/jsonschema/pointcloud_csv_to_mcap.py

```diff
@@ -0,0 +1,118 @@
+""" Example script for converting pointcloud CSV data to an MCAP viewable in Foxglove Studio.
+
+This script uses the public CSV dataset "Sydney Urban Objects Dataset",
+released by the Australian Centre for Field Robotics at the University of Sydney, NSW, Australia.
+The original dataset can be downloaded from here:
+https://www.acfr.usyd.edu.au/papers/SydneyUrbanObjectsDataset.shtml
+
+usage:
+    curl https://www.acfr.usyd.edu.au/papers/data/sydney-urban-objects-dataset.tar.gz | tar -x
+    pip install mcap
+    python3 pointcloud_csv_to_mcap.py sydney-urban-objects-dataset/objects/4wd.0.2299.csv -o 4wd.mcap
+"""
+import argparse
+import base64
+import csv
+import json
+import typing
+import struct
+from pathlib import Path
+import datetime
+
+# tutorial-mcap-imports-start
+from mcap.mcap0.writer import Writer
+from mcap.mcap0.well_known import SchemaEncoding, MessageEncoding
+
+# tutorial-mcap-imports-end
+
+# tutorial-csv-decode-start
+def point_reader(csv_path: typing.Union[str, Path]):
+    with open(csv_path, "r") as f:
+        for timestring, i, _, x, y, z, _, _, _ in csv.reader(f):
+            timestamp = datetime.datetime.strptime(timestring, "%Y%m%dT%H%M%S.%f")
+            yield (timestamp, float(i), float(x), float(y), float(z))
+            # tutorial-csv-decode-end
+
+
+def main():
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument("csv", help="The input CSV to read")
+    parser.add_argument(
+        "--output", "-o", default="out.mcap", help="The MCAP output path to write"
+    )
+    args = parser.parse_args()
+
+    pointcloud: typing.Dict[str, typing.Any]
```

Comment:
is this type annotation really doing much for us? is it necessary to be able to assign to new keys later?

Reply:
I did it to stop `pylance` annoying me when I added new keys later, yes.